app/models/favorite.rb

```diff
@@ -13,7 +13,26 @@ def add_pet(id)
     @contents[id.to_s] = count_of(id) + 1
   end
 
+  #if it's already favorited, dont add
+  #if its not already favorited, add pet
+  def already_favorited?(id)
+    if !@contents.has_key?(id.to_s)
+      add_pet(id)
+    end
+  end
+
   def count_of(id)
     @contents[id.to_s].to_i
   end
+
+  #{pet.id => 1} ....use id to find pet object
+  def favorite_pets
```

Comment:
For every pet.id, I have one in my favorites. 
Now use that id and get me the information for the pet.
This will be used to print in the favorites index.